src/abstraction-memory.c

```diff
@@ -0,0 +1,97 @@
+// SPDX-License-Identifier: LGPL-2.1-only
+/**
+ * Libcgroup abstraction layer for the memory controller
+ *
+ * Copyright (c) 2021-2025 Oracle and/or its affiliates.
+ * Author: Tom Hromatka <tom.hromatka@oracle.com>
+ */
+
+#include "abstraction-common.h"
+#include "abstraction-map.h"
+
+#include <libcgroup.h>
+#include <libcgroup-internal.h>
+
+#include <stdlib.h>
+#include <string.h>
+#include <unistd.h>
+#include <errno.h>
+#include <stdio.h>
+
+#define V1_NEG1_STR "-1"
+#define V1_MAX_STR "9223372036854771712"
+#define V2_MAX_STR "max"
+
+#define LL_MAX 24
```

Comment:
@drakenclimber Should `LL_MAX` match upper limit allowed by Kernel of 21 characters (including the '\0')? The Kernel would silently trim the excess trailing digits and update setting with `V1_MAX_STR` value but this will leave the user wondering what went wrong. Either we can leave there, mimicing the Kernel behavior or we would check the strlen and return error?

Reply:
Sounds good to em.  I'll make that change